src/hooks/useMessage.ts

```diff
@@ -1,10 +1,11 @@
 import { useState } from 'react';
+import { Message, MessageType, UseMessageReturnType } from 'types/hooks.types';
 
-const useMessage = () => {
+const useMessage = (): UseMessageReturnType => {
```

Comment:
I think that the array as a return value here is "difficult" to use, as we return quite a lot of stuff from this hook. IMHO, it would be better to return an object, so when using this hook, you can get only the stuff you need from it. also, typescript will make it super easy to know what is returned, so we won't need to go inside this file to see what does the hook return

Reply:
Thats right, I left it as array because it was initially array - but it definitely should be returned as object